glean-core/ios/Glean/Scheduler/GleanOperation.swift

```diff
@@ -0,0 +1,43 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/// Base class for other Glean Operations that contains the management of
+/// the executing and finished key-value coding/observers
+class GleanOperation: Operation {
+
+    private var _executing = false {
```

Comment:
Can you explain this block of code? It looks ... weird

Reply:
I second this

Reply:
This is something I just learned about `Operation`.  The `isExecuting` and `isFinished` and other member properties use [KVC](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/KeyValueCoding/index.html) (key-value coding) which provides indirect access to properties and [KVO](https://developer.apple.com/documentation/swift/cocoa_design_patterns/using_key-value_observing_in_swift) (key-value observing) to allow for event driven observation of these properties. 

This is how that signalling looks to trigger events that are hooked into this. A more in-depth explanation can be found in the docs for [`Operation`](https://developer.apple.com/documentation/foundation/operation)

Reply:
I added some comments to help clarify this